src/nvim/charset.c

```diff
@@ -1598,10 +1609,14 @@ int vim_islower(int c)
   return islower(c);
 }
 
-int vim_isupper(int c)
+/// Check that the character is upper-case
+///
+/// @param  c  character to check
+bool vim_isupper(int c)
+  FUNC_ATTR_CONST FUNC_ATTR_WARN_UNUSED_RESULT FUNC_ATTR_NONNULL_ALL
```

Comment:
@Pyrohh I usually ignore this, only considering the main sense of such attributes (specifically, what optimizations can be applied here). But @bfredl is right about dependence on global state.